src/plugins/safeSnap/components/HandleOutcomeUma.vue

```diff
@@ -307,26 +304,30 @@ const questionState = computed(() => {
     return QuestionStates.noTransactions;
 
   const ts = (Date.now() / 1e3).toFixed();
-  const { proposalEvent, proposalExecuted } = questionDetails.value;
+  const { proposalEvent, proposalExecuted, activeProposal } =
+    questionDetails.value;
 
   // If proposal has already been executed, prevents user from proposing again.
   if (proposalExecuted) return QuestionStates.completelyExecuted;
 
   // User can confirm vote results if not done already and there is no proposal yet.
-  if (!proposalEvent && !voteResultsConfirmed.value)
+  if (!activeProposal && !voteResultsConfirmed.value)
     return QuestionStates.waitingForVoteConfirmation;
 
   // Proposal can be made if it has not been made already and user confirmed vote results.
-  if (!proposalEvent && voteResultsConfirmed)
+  if (!activeProposal && voteResultsConfirmed)
     return QuestionStates.waitingForProposal;
 
-  // Proposal has been made and is waiting for liveness period to complete.
-  if (!proposalEvent.isExpired && !proposalEvent.isDisputed)
-    return QuestionStates.waitingForLiveness;
+  // Proposal can be deleted if it has been rejected.
+  if (proposalEvent.isDisputed && proposalEvent.resolvedPrice == 0)
+    return QuestionStates.proposalRejected;
 
   // If disputed, a proposal can be deleted to enable a proposal to be proposed again.
   if (proposalEvent.isDisputed) return QuestionStates.disputedButNotResolved;
```

Comment:
Won't this be hit in the case where it's disputed and resolved, but the price is _not_ 0? Or am I missing something?

Reply:
Disputed proposals are assumed to be invalid and have to be deleted in the OptimisticGovernor and re-proposed, so there isn't a scenario where you would wait for resolution and use the DVM-resolved price (and a resolution of 1 wouldn't help, anyway, since the OG detects the proposal was disputed and rejects it regardless).

Reply:
Just to elaborate a little bit, we check for the case where `resolvedPrice == 0` first only because the Optimistic Governor has a different deletion flow for rejected proposals. We should look into this, in case there's an edge case where a proposal is resolved to 1, to see if it can still be deleted and re-proposed.